ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/schema/internal/SchemaServiceImpl.java

```diff
@@ -79,31 +95,47 @@ public SchemaSpace getSchemas(SchemaSpaceID spaceID) {
 		Preconditions.checkNotNull(spaceID);
 
 		synchronized (spaces) {
-			DefaultSchemaSpace space = spaces.get(spaceID);
+			DefaultResourceSchemaSpace space = spaces.get(spaceID);
 			if (space == null) {
-				space = new DefaultSchemaSpace();
+				space = new DefaultResourceSchemaSpace();
 				spaces.put(spaceID, space);
 			}
 			return space;
 		}
 	}
 
 	/**
-	 * @see SchemaService#addSchema(Schema, SchemaSpaceID)
+	 * @see eu.esdihumboldt.hale.ui.service.schema.SchemaService#addSchema(java.lang.String,
+	 *      eu.esdihumboldt.hale.common.schema.model.Schema,
+	 *      eu.esdihumboldt.hale.common.schema.SchemaSpaceID)
 	 */
 	@Override
-	public void addSchema(Schema schema, SchemaSpaceID spaceID) {
+	public void addSchema(String resourceID, Schema schema, SchemaSpaceID spaceID) {
 		Preconditions.checkNotNull(spaceID);
 
 		SchemaIO.loadMappingRelevantTypesConfig(schema, spaceID, getProjectService()
 				.getConfigurationService());
-
-		DefaultSchemaSpace space = (DefaultSchemaSpace) getSchemas(spaceID);
-		space.addSchema(schema);
-
+		
+		DefaultResourceSchemaSpace space = (DefaultResourceSchemaSpace) getSchemas(spaceID);
+		space.addSchema(resourceID, schema);
+		
 		notifySchemaAdded(spaceID, schema);
 	}
 
+	/**
+	 * @see eu.esdihumboldt.hale.ui.service.schema.SchemaService#removeSchema(java.lang.String,
+	 *      eu.esdihumboldt.hale.common.schema.SchemaSpaceID)
+	 */
+	@Override
+	public void removeSchema(String resourceId, SchemaSpaceID spaceID) {
+		Preconditions.checkNotNull(spaceID);
+		
+		DefaultResourceSchemaSpace space = (DefaultResourceSchemaSpace) getSchemas(spaceID);
+		space.removeSchema(resourceId);
+		notifySchemasCleared(spaceID);
```

Comment:
Would it maybe make sense to introduce a new type of event for the schema removal?

Reply:
A new event `notifySchemaRemoved(SchemaSpaceID spaceID)` is created and also added a listener  `schemaRemoved(SchemaSpaceID spaceID);` for (any) future requirements.